OpenRA.Mods.RA/Attack/AttackWander.cs

```diff
@@ -8,56 +8,30 @@
  */
 #endregion
 
-using OpenRA.Activities;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.RA.Traits
 {
 	[Desc("Will AttackMove to a random location within MoveRadius when idle.",
 		"This conflicts with player orders and should only be added to animal creeps.")]
-	class AttackWanderInfo : ITraitInfo, Requires<AttackMoveInfo>
+	class AttackWanderInfo : WandersInfo, Requires<AttackMoveInfo>
 	{
-		public readonly int WanderMoveRadius = 10;
-
-		[Desc("Number of ticks to wait until decreasing the effective move radius.")]
-		public readonly int MoveReductionRadiusScale = 5;
-
-		public object Create(ActorInitializer init) { return new AttackWander(init.self, this); }
+		public override object Create(ActorInitializer init) { return new AttackWander(init.self, this); }
 	}
 
-	class AttackWander : INotifyIdle
+	class AttackWander : Wanders
 	{
 		readonly AttackMove attackMove;
-		readonly AttackWanderInfo info;
-
-		int ticksIdle;
-		int effectiveMoveRadius;
 
 		public AttackWander(Actor self, AttackWanderInfo info)
+			: base(self, info)
 		{
-			this.info = info;
-			effectiveMoveRadius = info.WanderMoveRadius;
 			attackMove = self.TraitOrDefault<AttackMove>();
 		}
 
-		public void TickIdle(Actor self)
+		public override void DoAction(Actor self, CPos targetPos)
 		{
-			var target = self.CenterPosition + new WVec(0, -1024 * effectiveMoveRadius, 0).Rotate(WRot.FromFacing(self.World.SharedRandom.Next(255)));
-			var targetCell = self.World.Map.CellContaining(target);
-
-			if (!self.World.Map.Contains(targetCell))
-			{
-				// If MoveRadius is too big there might not be a valid cell to order the attack to (if actor is on a small island and can't leave)
-				if (++ticksIdle % info.MoveReductionRadiusScale == 0)
-					effectiveMoveRadius--;
-
-				return;  // We'll be back the next tick; better to sit idle for a few seconds than prolong this tick indefinitely with a loop
-			}
-
-			attackMove.ResolveOrder(self, new Order("AttackMove", self, false) { TargetLocation = targetCell });
-
-			ticksIdle = 0;
-			effectiveMoveRadius = info.WanderMoveRadius;
+			attackMove.ResolveOrder(self, new Order("AttackMove", self, false) { TargetLocation = targetPos });
```

Comment:
OK. But who's going to resolve the order then? Because is sure won't be `readonly AttackMove attackMove;`